roles/couchbase-infra/tasks/main.yml

```diff
@@ -182,9 +162,9 @@
       metadata:
         name: rh-catalog
         namespace: "{{ couchbase_namespace }}"
-      type: kubernetes.io/dockerconfigjson
       data:
-        .dockerconfigjson: "{{ rawdockerconfigjson | to_json | b64encode }}"
+        .dockerconfigjson: "{{ couchbase.pull_secret }}"
```

Comment:
I'm good with this for now. It's definitely better than my previous approach. We'll need to add some documentation, so I'll make README edits.